src/acrcssc/azext_acrcssc/_validators.py

```diff
@@ -0,0 +1,183 @@
+            if task is None:
+                missing_tasks.append(task_name)
+            else:
+                task_list.append(task)
+
+        if len(missing_tasks) > 0:
+            logger.debug(f"Failed to find tasks {', '.join(missing_tasks)} from registry {registry.name}")
+            return False, task_list
+
+        return True, task_list
+    except Exception as exception:
+        logger.debug(f"Failed to find tasks from registry {registry.name} : {exception}")
+        return False, task_list
+
+
+def check_continuous_task_config_exists(cmd, registry):
+    # A client cannot be used in this situation because the 'show registry/image'
+    # is a data plane operation and the az cli does not include the data plane API.
+    subscription = parse_resource_id(registry.id)[SUBSCRIPTION]
+    try:
+        token = _get_acr_token(registry.name, subscription)
+        acr_repository_show(
+            cmd=cmd,
+            registry_name=registry.name,
+            repository=f"{CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}",
+            username=BEARER_TOKEN_USERNAME,
+            password=token)
+    except Exception as exception:
+        if hasattr(exception, 'status_code') and exception.status_code == 404:
+            return False
+        # report on the error only if we get something other than 404
+        logger.debug(f"Failed to find config {CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG} from registry {registry.name} : {exception}")
+        raise
+    return True
+
+
+def _validate_schedule(schedule):
+    # during update, schedule can be null if we are only updating the config
+    if schedule is None:
+        return
+    # Extract the numeric value and unit from the timespan expression
+    match = re.match(r'(\d+)(d)$', schedule)
+    if not match:
+        raise InvalidArgumentValueError(error_msg=ERROR_MESSAGE_INVALID_TIMESPAN_FORMAT, recommendation=RECOMMENDATION_SCHEDULE)
+
+    value = int(match.group(1))
+    unit = match.group(2)
+    if unit == 'd' and (value < 1 or value > 30):  # day of the month
+        raise InvalidArgumentValueError(error_msg=ERROR_MESSAGE_INVALID_TIMESPAN_VALUE, recommendation=RECOMMENDATION_SCHEDULE)
+
+
+def validate_inputs(schedule, config_file_path=None, dryrun=False, run_immediately=False):
+    _validate_schedule(schedule)
+    if config_file_path is not None:
+        validate_continuouspatch_config_v1(config_file_path)
+    validate_run_type(dryrun, run_immediately)
+    
+
+def validate_run_type(dryrun, run_immediately):
+    if dryrun and run_immediately:
+        raise InvalidArgumentValueError(error_msg="The --dryrun and --run-immediately options cannot be used together. Use one or the other.")
+
+
+def validate_task_type(task_type):
+    if (task_type not in [item.value for item in CSSCTaskTypes]):
+        raise InvalidArgumentValueError(error_msg=ERROR_MESSAGE_INVALID_TASK)
```

Comment:
Suggestion: Improve the error message so the customer knows the current type and the expected valid types.